python_modules/dagster/dagster_tests/core_tests/storage_tests/test_io_manager.py

```diff
@@ -757,3 +759,66 @@ def my_job():
         my_op2(my_op())
 
     assert my_job.execute_in_process(partition_key="2020-01-01").success
+
+
+def test_context_logging_user_events():
+    class DummyIOManager(IOManager):
+        def __init__(self):
+            self.values = {}
+
+        def handle_output(self, context, obj):
+            keys = tuple(context.get_output_identifier())
+            self.values[keys] = obj
+
+            context.log_event(AssetMaterialization(asset_key="first"))
+            time.sleep(1)
+            context.log.debug("foo bar")
+            context.log_event(AssetMaterialization(asset_key="second"))
```

Comment:
oh its just the implicit return value of this being the last statement of the function - that seems dubious. Were double yielding that event then.